workflow/controller/controller_test.go

```diff
@@ -513,6 +484,43 @@ func syncPodsInformer(ctx context.Context, woc *wfOperationCtx, podObjs ...apiv1
 	}
 }
 
+func withOutputs(outputs wfv1.Outputs) with {
```

Comment:
nit: could you put these in-line with `with` and the old functions / above `createRunningPods`? i.e. so that the diff lines up 😅